GVFS/GVFS.Common/GVFSEnlistment.cs

```diff
@@ -79,6 +82,24 @@ public string GVFSHooksVersion
             get { return this.gvfsHooksVersion; }
         }
 
+        public string EnlistmentId
+        {
+            get
+            {
+                this.InitializeIds();

```

Comment:
I don't think it's a good idea to have a property do things as expensive as shelling out to a process.

Is there a good reason to lazy-initialize these values? Can we just initialize them up front and get rid of this extra complexity?

Reply:
@kewillford please do correct me if I'm wrong

GVFSVerb.ForExistingEnlistment will create a GVFSEnlistment, but I"m not seeing any of the verbs that extend it use enlistmentId.  Ex Dehydrate/Diagnose/Status/Prefetch.  I do think it would be great if we have the value there though!

MountId gets a little bit more tricky.  We don't need it for Repair for instance, but probably not a killer if we look up another config setting.  However, MountId can change from it's previous config value while we run the mount process.  I think making this one into a function instead of a property and looking it up on-demand is probably safest.  It should at most be called once and that would match the current functionality.

Reply:
Currently only the CloneVerb and MountVerb are calling `LogEnlistmentInfoAndSetConfigValues` that initialized the EnlistmentId and MountId and sets them in the config.  Later code was added in `CreateTracer` of the InProcessMountVerb to lookup up the ids so that the tracer that gets created could always have those for any ETW event.  Having them as a part of the Enlistment makes sense and should then be available when needed but in some cases the Tracer is created before any Enlistment so we should check those cases and see if there is something we need to do there.

Reply:
In most cases we only have one Enlistment object that gets created and what if at some point we want `gvfs status` to output the EnlistmentId and MountId so we can ask a user for them and run queries against them?

Reply:
@kewillford that's for the clarification here!  I've updated it to be a lookup on demand which is hopefully pretty straightforward.  Since it's on Enlistment it's a pretty central place to get at EnlistmentId/MountId.  

My understanding (and @derrickstolee  feel free to correct me) is that we want status to be as fast as possible so a config lookup wouldn't be in the cards.

Reply:
I think `gvfs status` is ok to slow down a bit _if we actually show the information_. Kevin knows this area the best, so I defer to his opinions.

Reply:
I think we can defer initializing in the constructor until we need them initialized in more cases.

Reply:
I really do appreciate the conversation @kewillford / @derrickstolee !  At a minimum now we do have an easy way to grab enlistment/mountIds.  

I will look at adding them to more verbs when we have the need.